hibernate-reactive-core/src/main/java/org/hibernate/reactive/stage/impl/StageSessionFactoryImpl.java

```diff
@@ -171,6 +179,7 @@ public <T> CompletionStage<T> withSession(String tenantId, Function<Stage.Sessio
 		else {
 			LOG.debugf( "No existing open Stage.Session was found in the current Vert.x context for current tenant '%s': opening a new instance", tenantId );
 			return withSession( openSession( tenantId ), work, key );
+			return executeLazily( v -> withSession( openSession( tenantId ), work, key ) );
```

Comment:
Like my previous comment, I suspect these various `executeLazily` are not acually necessary. Same for the other instances.